### DISCUSSION TITLE
ESP8266 + ILI9341 - readRegister() / readRect() failed

### DISCUSSION THREAD

**Discussion by sfxsfl** [Bodmer/TFT_eSPI#2704] [Q&A - General]
Hi Bodmer,
once again kudos for you and your great work on TFT_eSPI. It's working amazingly fast and robust.

Currently I'm trying to use readRect in conjunction with pushRect function to save a small area
on my display - unfortunately without success. 

I've tried to use the readRegister() example from one of the provided examples with the following output. 

---- Register ----
Register 0x04: 0x00000000
Register 0x09: 0xFFFFFFFF
Register 0x0A: 0x000000FF
Register 0x0B: 0x00000000
Register 0x0C: 0x00000000
Register 0x0F: 0x000000FF
Register 0x2E: 0x00000000
Register 0xDA: 0x00000000
Register 0xDB: 0x00000000
Register 0xDC: 0x00000000
Register 0xDD: 0x000000FF
Register 0xD3: 0x00FFFFFF

So I'll guess it's a problem reading values from the ILI9341 back. Any suggestion if I have missed
something ? Is this a know issue, while using TFT_SPI_OVERLAP ? 

I'm using the following configuration in my user_setup.h 

`#define ILI9341_DRIVER       // Generic driver for common displays`
`#define TFT_SPI_OVERLAP`

`// In ESP8266 overlap mode the TFT chip select MUST connect to pin D3`
`#define TFT_CS          PIN_D3`
`#define TFT_DC         PIN_D1`
`#define TOUCH_CS   PIN_D4  // Chip select pin (T_CS) of touch screen`

Thanks in advance 
Stefan

![grafik](https://github.com/Bodmer/TFT_eSPI/assets/7534354/b0ba8d8e-30e6-4254-a541-242afd3c42d9)

**Comment by sfxsfl** [comment]
![grafik](https://github.com/Bodmer/TFT_eSPI/assets/7534354/961122b3-a139-4193-a863-10ef4f1d5050)

**Comment by sfxsfl** [comment]
Maybe the problem is related to the wiring and the related CS Signals. Help will be highly appreciated ... Thanks in advance Stefan

**Comment by sfxsfl** [comment]
Has anyone tried to read Display registers while using SPI Overlap mode ??? I'm still struggling with this issue ...

Thanks in advance
Stefan

**Reply by Bodmer** [reply]
Unfortunately reads in overlap mode are not supported by the library. It may be posible to add this but lack of time means I am not able to do this:
https://github.com/Bodmer/TFT_eSPI/blob/master/Processors/TFT_eSPI_ESP8266.c#L43

**Reply by sfxsfl** [reply]
Hi Bodmer, thanks a lot for the swift reply and I fully understand that you won't have time for the implementation. I have a one final question on this topic. Does that mean _**uint8_t TFT_eSPI::readcommand8(uint8_t cmd_function, uint8_t index)**_ won't work, too ? 

Thanks a lot for the clarification in advance
Stefan
